validation/src/extension.ts

```diff
@@ -0,0 +1,115 @@
+import { ExtensionContext, languages, Disposable, workspace, window } from 'vscode';
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { spawn } from 'child_process';
+
+let disposables: Disposable[] = [];
+
+export function activate(context: ExtensionContext) {
+  const collection = languages.createDiagnosticCollection('vela');
+  if (window.activeTextEditor) {
+    updateDiagnostics(window.activeTextEditor.document, collection);
+  }
+
+  context.subscriptions.push(workspace.onDidChangeTextDocument(documentEvent => {
+    if (documentEvent) {
+      updateDiagnostics(documentEvent.document, collection);
+    }
+  }));
+
+  context.subscriptions.push(window.onDidChangeActiveTextEditor(editor => {
+    if (editor) {
+      updateDiagnostics(editor.document, collection);
+    }
+  }));
+}
+
+function runVelaVet(document: vscode.TextDocument): Promise<string> {
+  const command = `vela def vet ${document.fileName}`;
+  const process = spawn(command, { shell: true });
+
+  return new Promise((resolve, reject) => {
+    process.stdout.on('data', (data) => {
+      resolve(data.toString());
+    });
+
+    process.stderr.on('data', (data) => {
+      reject(data.toString());
+    });
+  });
+}
+
+function determineKeyword(problem: string): string | undefined {
+  const regexes = [
+    // Error: failed to parse CUE: /Users/kuba/personal_repos/vscode-plugin/validation/examples/dummy.cue: invalid type trit
+    /invalid type (.+)/,
+    // Error: failed to parse CUE: /Users/kuba/personal_repos/vscode-plugin/validation/examples/dummy.cue: invalid definition spec: json: cannot unmarshal number into Go struct field TraitDefinitionSpec.podDisruptive of type bool
+    // Error: failed to parse CUE: /Users/kuba/personal_repos/vscode-plugin/validation/examples/dummy.cue: invalid definition spec: json: cannot unmarshal number into Go struct field WorkloadTypeDescriptor.workload.type of type string
+    /field [\w\.]+\.+(\w+) of type/,
+    // Error: failed to parse CUE: /Users/kuba/personal_repos/vscode-plugin/validation/examples/dummy.cue: test2.attributes.podDisruptive: reference "tru" not found
+    /reference \"(.+)\" not found/,
+    // Error: failed to parse CUE: /Users/kuba/personal_repos/vscode-plugin/validation/examples/dummy.cue: invalid definition spec: json: unknown field "podDisruptive"
+    /unknown field "(.+)"/
+  ];
```

Comment:
ideally, all of this logic id moved over to `vela def vet` for it to report line and column numbers accurately.